instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/Metric.java

```diff
@@ -61,17 +66,30 @@ MetricInfo buildMetricInfo(
 
     MetricInfo.Type metricType = getMetricType();
     if (metricType == null) {
-      metricType = defaultType;
+      metricType = jmxRule.getMetricType();
+    }
+    if (metricType == null) {
+      metricType = MetricInfo.Type.GAUGE;
```

Comment:
[for reviewer] The same default is applied in `MetricInfo` constructor, moving it here makes it happen a bit earlier. I expect a later refactor of this would bring the complete building and validation logic into this method.